### PR TITLE
[sparsity m/n] sparsity mask m/n stays constant during the fine-tuning

### PR DESCRIPTION
**Pull request by wwwind**
Corrected implementation of sparsity m/n, so that we create mask at the first step and it stays the same during the fine-tuning.

### CODE REVIEW

**Review by rino20 — APPROVED** [review]

**Comment by rino20** [comment]
https://github.com/tensorflow/model-optimization/blob/master/tensorflow_model_optimization/python/core/sparsity/keras/prune_integration_test.py

is failing.

**Comment by wwwind** [comment]
Hi @rino20 I removed forcing to the eager mode and this caused this failure. https://github.com/tensorflow/model-optimization/pull/847/commits/fe5699f536a673adbfd5b4a20e5f62f8ca652bce
it looks that the fix is working only for eager mode. i am looking at how to correct it.

**Comment by rino20** [comment]
Thanks, I will wait for the fix.

**Comment by wwwind** [comment]
I am closing this PR as there is a replacement here: https://github.com/tensorflow/model-optimization/pull/860
